### PR TITLE
(FACT-2029) Solaris release facts

### PR DESCRIPTION
**Pull request by alexandru-d-popa**

### CODE REVIEW

File: spec/facter/resolvers/solaris/solaris_release_resolver.rb [hunk 1]
```diff
@@ -0,0 +1,27 @@
+# frozen_string_literal: true
+
+describe 'SolarisOsReleaseResolver' do
```
Comment (sebastian-miclea) [inline spec/facter/resolvers/solaris/solaris_release_resolver.rb]: I don't think these specs were run.

File: lib/resolvers/solaris/solaris_os_release_resolver.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+module Facter
+  module Resolvers
+    class SolarisOsRelease < BaseResolver
```
Comment (sebastian-miclea) [inline lib/resolvers/solaris/solaris_os_release_resolver.rb]: Could you rename this class to **SolarisRelease**? The direction we're heading towards is to have the resolver name describe as much as possible on what it checks, like _lsb_release_resolver_.

File: lib/resolvers/solaris/solaris_os_release_resolver.rb [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            subscribe_to_manager
+            result || build_release_facts(fact_name)
+          end
+        end
+
+        private
+
+        def build_release_facts(fact_name)
+          result = read_os_release_file
+          return nil if result.nil?
```
Comment (oanatmaria) [inline lib/resolvers/solaris/solaris_os_release_resolver.rb]: just return

File: lib/resolvers/solaris/solaris_os_release_resolver.rb [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+          result = text.match(regex_pattern)
+          major, minor = result.captures if result
+          return [major, minor] if major && minor
+        end
+
+        def read_os_release_file
+          output, error = Open3.capture2('cat /etc/release')
+          if error
+            @log.error('Could not build release fact because of missing file /etc/release')
+            return nil
```
Comment (oanatmaria) [inline lib/resolvers/solaris/solaris_os_release_resolver.rb]: just return

File: spec/facter/resolvers/solaris/solaris_release_resolver_spec.rb [hunk 5]
```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+describe 'SolarisOsRelease' do
+  before do
+    status = double(Process::Status, to_s: st)
+    allow(Open3).to receive(:capture2)
+      .with('cat /etc/release')
+      .and_return([output, status])
+    # allow_any_instance_of(Process::Status).to receive(:to_s).and_return(st)
```
Comment (oanatmaria) [inline spec/facter/resolvers/solaris/solaris_release_resolver_spec.rb]: delete commented line

**Review by sebastian-miclea — APPROVED** [review]

**Pull request merged**